sdk/go/common/util/yamlutil/edit.go

```diff
@@ -0,0 +1,125 @@
+package yamlutil
+
+import (
+	"bytes"
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+)
+
+type HasRawValue interface {
+	RawValue() []byte
+}
+
+// Edit does a deep comparison on original and new and returns a YAML document that modifies only
+// the nodes changed between original and new.
+func Edit(original []byte, new interface{}) ([]byte, error) {
+	var err error
+	var oldDoc yaml.Node
+	err = yaml.Unmarshal(original, &oldDoc)
+	if err != nil {
+		return nil, err
+	}
+
+	newBytes, err := yaml.Marshal(new)
+	if err != nil {
+		return nil, err
+	}
+	var newValue yaml.Node
+	err = yaml.Unmarshal(newBytes, &newValue)
+	if err != nil {
+		return nil, err
+	}
+
+	newValue, err = editNodes(&oldDoc, &newValue)
+	if err != nil {
+		return nil, err
+	}
+
+	var b bytes.Buffer
+	yamlEncoder := yaml.NewEncoder(&b)
+	yamlEncoder.SetIndent(2)
+	err = yamlEncoder.Encode(&newValue)
+	if err != nil {
+		return nil, fmt.Errorf("error editing value %#v: %w", newValue, err)
+	}
+
+	return b.Bytes(), nil
```

Comment:
If there was a way to avoid repeating this in `sdk/go/common/encoding/marshal.go`, that would be great.

Reply:
Yeah I think so - we could modify an integration test to start with a file that has some config and comments.

Reply:
Exported a two space indendation `YamlEncode` and refactored.